huaweicloud/services/live/resource_huaweicloud_live_disable_push_stream.go

```diff
@@ -0,0 +1,279 @@
+package live
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/hashicorp/go-multierror"
+	"github.com/hashicorp/go-uuid"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+
+	"github.com/chnsz/golangsdk"
+
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/common"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/config"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/utils"
+)
+
+// @API Live POST /v1/{project_id}/stream/blocks
+// @API Live GET /v1/{project_id}/stream/blocks
+// @API Live PUT /v1/{project_id}/stream/blocks
+// @API Live DELETE /v1/{project_id}/stream/blocks
+func ResourceDisablePushStream() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceDisablePushStreamCreate,
+		ReadContext:   resourceDisablePushStreamRead,
+		UpdateContext: resourceDisablePushStreamUpdate,
+		DeleteContext: resourceDisablePushStreamDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: resourceDisablePushStreamImportState,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"region": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+				ForceNew: true,
+			},
+			"domain_name": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: `Specifies the ingest domain name of the disable push stream.`,
+			},
+			"app_name": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: `Specifies the application name of the disable push stream.`,
+			},
+			"stream_name": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: `Specifies the stream name of the disable push stream.`,
+			},
+			"resume_time": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Computed:    true,
+				Description: `Specifies the time of the resuming push stream.`,
+			},
+		},
+	}
+}
+
+func resourceDisablePushStreamCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	var (
+		cfg    = meta.(*config.Config)
+		region = cfg.GetRegion(d)
+	)
+
+	client, err := cfg.NewServiceClient("live", region)
+	if err != nil {
+		return diag.Errorf("error creating Live client: %s", err)
+	}
+
+	createHttpUrl := "v1/{project_id}/stream/blocks"
+	createPath := client.Endpoint + createHttpUrl
+	createPath = strings.ReplaceAll(createPath, "{project_id}", client.ProjectID)
+
+	createOpt := golangsdk.RequestOpts{
+		KeepResponseBody: true,
+		OkCodes: []int{
+			204,
+		},
+		JSONBody: utils.RemoveNil(buildDisablePushStreamBodyParams(d)),
+	}
+
+	_, err = client.Request("POST", createPath, &createOpt)
+	if err != nil {
+		return diag.Errorf("error creating disable push stream: %s", err)
```

Comment:
```suggestion
		return diag.Errorf("error creating disabled push stream: %s", err)
```
Please check other similar questions elsewhere.